src/harness/util.py

```diff
@@ -408,3 +409,51 @@ def filterChannelsByFrequencyRange(channels, freq_range):
       and
       channel['frequencyRange']['highFrequency'] <= freq_range['highFrequency']
   ]
+
+def _orderAttributes(obj):
+    if isinstance(obj, dict):
+        return sorted((k, _orderAttributes(v)) for k, v in obj.items())
+    if isinstance(obj, list):
+        return sorted(_orderAttributes(x) for x in obj)
+    else:
+        return obj
+
+def compareDictWithUnorderedLists(first_dict, second_dict):
+  """ Deep comparison of two dictionaries
+
+  Args:
+    first_dict: first dictionary to be compared.
+    second_dict: second dictionary to be compared.
+ Returns: boolean set to true if the dictionaries are equal and false otherwise.
+  """
+  return _orderAttributes(first_dict) == _orderAttributes(second_dict)
+
+def areTwoPpasEqual(first_ppa, second_ppa, delta=10):
+  """ Deep comparison of two PPAs considering
+
+  Args:
+    first_ppa: a dictionary contains a reference of PPA data to compare to,
+               the points' order of the ploygon of this PPA is used a reference.
+    second_ppa: a dictionary contains PPA data to be compared.
+    delta: an accepted margin for comparing the polygons of PPAs
+
+ Returns: boolean set to true if the PPAs are equal and false otherwise.
+ Note: the PPA should contain only one GeoJson Feature, this feature should contain only one Polygon
+       and polygons with holes are accepted
+  """
+  geometry_of_first_ppa = first_ppa['zone']['features'][0]['geometry']
+  geometry_of_second_ppa = second_ppa['zone']['features'][0]['geometry']
+
+  if not utils.PolygonsAlmostEqual(geometry_of_first_ppa, geometry_of_second_ppa, delta):
+    return False
+  if first_ppa['zone']['features'][0]['geometry']['type'] \
```

Comment:
You can use:` geometry_of_first_ppa['type'] here, instead of the full dict reindexing

Reply:
yes, thank you I upodated it.